Modules/Quotas/src/Quotas.ConsumerApi/Controllers/QuotasController.cs

```diff
@@ -0,0 +1,37 @@
+﻿using Backbone.BuildingBlocks.API;
+using Backbone.BuildingBlocks.API.Mvc;
+using Backbone.BuildingBlocks.Application.Abstractions.Exceptions;
+using Backbone.BuildingBlocks.Application.Pagination;
+using Backbone.Modules.Quotas.Application;
+using Backbone.Modules.Quotas.Application.DTOs;
+using Backbone.Modules.Quotas.Application.Identities.Queries.GetQuotasForIdentity;
+using MediatR;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Options;
+using ApplicationException = Backbone.BuildingBlocks.Application.Abstractions.Exceptions.ApplicationException;
+
+namespace Backbone.Modules.Quotas.ConsumerApi.Controllers;
+
+[Route("api/v1/[controller]")]
+[Authorize("OpenIddict.Validation.AspNetCore")]
+public class QuotasController : ApiControllerBase
+{
+    private readonly ApplicationOptions _options;
+
+    public QuotasController(IMediator mediator, IOptions<ApplicationOptions> options) : base(mediator)
+    {
+        _options = options.Value;
+    }
+
+    [HttpGet("ListIndividualQuotas")]
+    [ResponseCache(Duration = 1800, Location = ResponseCacheLocation.Any, VaryByQueryKeys = new string[] { "address" })]
+    [ProducesResponseType(typeof(List<QuotaDTO>), StatusCodes.Status200OK)]
```

Comment:
Did you copy this from somewhere? If so, can you please tell me from where? Because it's incorrect. Look at the `MessagesController` for example, to see how it should look like.

Reply:
I've consulted the docs you provided: [https://learn.microsoft.com/en-us/aspnet/core/performance/caching/response?view=aspnetcore-8.0](https://learn.microsoft.com/en-us/aspnet/core/performance/caching/response?view=aspnetcore-8.0)

There is this piece of code:
```csharp
[Route("api/[controller]/ms")]
[HttpGet]
[ResponseCache(Duration = 10, Location = ResponseCacheLocation.Any, NoStore = false)]
public ContentResult GetTimeMS() => Content(
                  DateTime.Now.Millisecond.ToString());
```

I increased the duration and added the `VaryByQueryKeys` parameter to distinguish between different users, though it wasn't working as expected initially (it has been fixed, or at least an attempt to fix it has been made now).

Reply:
This comment was referring to the `ProducesResponseType` attribute. Not the `ResponseCache` one.

Reply:
I misunderstood decorations parameters like this:
```csharp
...typeof(HttpResponseEnvelopeResult<List<QuotaDTO>>)...
```

Fixed now.